### DISCUSSION TITLE
Automating JS prompt with two inputs

### DISCUSSION THREAD

**Discussion by Jegomezre** [webdriverio/webdriverio#12617] [Q&A]
I want to automate a login flow that prompts what is shown in the image attached, using webdriverio. Any ideas on how i could approach this or what is the path to take? Been stuck a while.
![image](https://github.com/webdriverio/webdriverio/assets/92931023/f4f6c2d6-de88-4f33-becb-c096456a2f38)

**Comment by portyu9** [comment]
Hi @Jegomezre 

A real JavaScript `window.prompt()` only has **one** text input, so if the screenshot shows a username + password dialog it is probably not a JavaScript prompt.

If it is the browser's **HTTP Basic Authentication** dialog, don't try to locate or type into that browser UI. Current WebdriverIO supports authentication as part of navigation:

```js
await browser.url('https://example.com/protected', {
    auth: {
        user: process.env.TEST_USER,
        pass: process.env.TEST_PASSWORD
    }
})
```

If it is actually an HTML login modal rendered by the application, then locate its two inputs normally with `$()` and use `setValue()`.

If it really is a JavaScript prompt, WebDriver can only send one string to it because a JS prompt itself only exposes one input.

The screenshot attached to the original discussion could not be retrieved from the indexed GitHub page, so I can't safely state which of those three cases it is. Based on the described "two inputs", I would check for HTTP authentication first.

There were no replies.

Please mark this answer as accepted if it helped, thank you.
